File: php/testcode/benchmark_test_00128.php
<?php
require_once __DIR__ . '/shared.php';

function benchmarkTest00128(BenchmarkRequest $req): BenchmarkResponse {
    $apiKey = $req->cookie('api_key');
    if (empty($apiKey) || !validateApiKey($apiKey)) {
        return BenchmarkResponse::error('unauthorized');
    }
    $result = performApiAction($req->param('action'));
    return BenchmarkResponse::json(['result' => $result]);
}

File: rust/testcode/benchmark_test_00128.rs
fn db_get_document(id: &str) -> String {
    format!("document_content_for_{}", id)
}

pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let id = req.param("id");
    let doc = db_get_document(&id);
    super::shared::BenchmarkResponse::ok(&doc)
}
